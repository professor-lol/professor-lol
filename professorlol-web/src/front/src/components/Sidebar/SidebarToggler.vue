<template>
  <button
    :display="display"
    :mobile="mobile"
    :class="classList"
    type="button"
    @click="sidebarToggle">
    <span class="navbar-toggler-icon" />
  </button>
</template>

<script>
import { sidebarCssClasses, validBreakpoints, checkBreakpoint } from '../../shared/classes'
import toggleClasses from '../../shared/toggle-classes'

export default {
  name: 'SidebarToggler',
  props: {
    defaultOpen: {
      type: Boolean,
      default: true
    },
    display: {
      type: String,
      default: 'lg'
    },
    mobile: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classList () {
      return [
        'navbar-toggler'
      ]
    }
  },
  mounted: function () {
    this.toggle(this.defaultOpen)
  },
  methods: {
    toggle (force) {
      const [display, mobile] = [this.display, this.mobile]
      let cssClass = sidebarCssClasses[0]
      if (!mobile && display && checkBreakpoint(display, validBreakpoints)) {
        cssClass = `sidebar-${display}-show`
      }
      toggleClasses(cssClass, sidebarCssClasses, force)
    },
    sidebarToggle (e) {
      e.preventDefault()
      e.stopPropagation()
      this.toggle()
    }
  }
}
</script>
