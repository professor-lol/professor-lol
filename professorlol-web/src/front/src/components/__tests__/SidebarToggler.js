import { mount } from "@vue/test-utils";
import SidebarToggler from "../Sidebar/SidebarToggler";

describe("SidebarToggler.vue", () => {
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = mount(SidebarToggler, {
      propsData: {
        defaultOpen: true,
        display: 'lg',
        mobile: false
      }
    })
    expect(wrapper.props().defaultOpen).toBe(true)
    expect(wrapper.props().display).toEqual('lg')
    expect(wrapper.props().mobile).toBe(false)
  });
  it('has toggle method', () => {
    expect(typeof SidebarToggler.methods.toggle).toBe('function')
  })
  it('has sidebarToggle method', () => {
    expect(typeof SidebarToggler.methods.sidebarToggle).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarToggler)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.is('button')).toBe(true)
    expect(wrapper.classes()).toContain('navbar-toggler')
  })
});
