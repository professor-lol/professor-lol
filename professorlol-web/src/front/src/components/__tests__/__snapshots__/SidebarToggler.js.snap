// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarToggler.vue renders correctly 1`] = `
<button
  class="navbar-toggler"
  display="lg"
  type="button"
>
  <span
    class="navbar-toggler-icon"
  />
</button>
`;
